aws/cognito_user_pool/variables.tf

```diff
@@ -74,3 +74,25 @@ variable "groups" {
   ))
   default = {}
 }
+
+variable "create_user_messaging" {
+  description = "Provides basic customized messaging for new accounts"
+  type = object(
+    {
+      email_subject = optional(string)
+      email_message = optional(string)
+      sms_message   = optional(string)
+    }
+  )
+  default = {
+    email_subject = "Your temporary password"
+    email_message = <<-EOT
+      Your username is {username} and temporary password is
+      <strong>{####}</strong>
+    EOT
+    sms_message   = <<-EOT
+      Your username is {username} and temporary password is
+      <strong>{####}</strong>
+    EOT
+  }
```

Comment:
This makes it so that in order to override the default, you need to provide all three values, which I'm guessing isn't the intension. Two options to resolve this:

- Instead of using `default = "..."` within the variable definition, create a local value inside of `aws/cognito_user_pool/main.tf` that uses the `defaults` function to provide the default values (see https://www.terraform.io/docs/language/functions/defaults.html)
- Break it into three separate variables of type `string` with a `default = "..."` value for each one. Alternatively, you could separate email and sms into two separate variables and just remove the optionals

Reply:
Yeah, that's actually a better approach. I opted for the first suggestion, which I think worked well.